dexcom/fetch/runner.go

```diff
@@ -399,9 +399,19 @@ func (t *TaskRunner) updateDataSet(dataSetUpdate *data.DataSetUpdate) error {
 }
 
 func (t *TaskRunner) fetchSinceLatestDataTime() error {
-	startTime := initialDataTime
+
+	var startTime time.Time
+	var err error
 	if t.dataSource.LatestDataTime != nil && startTime.Before(*t.dataSource.LatestDataTime) {
 		startTime = *t.dataSource.LatestDataTime
+	} else {
+		startTime, err = t.fetchDataRangeStart()
```

Comment:
It occurs to me that we could further utilize the dataRange API response, since it also returns the end times. We could use that to avoid making the subsequent calls, since we could already know there's no new data to fetch.